cocos/serialization/deserialize.ts

```diff
@@ -481,13 +450,37 @@ interface IFileData extends Array<any> {
     [File.DependUuidIndices]: (StringIndex|string)[];
 }
 
-// type Body = Pick<IFileData, File.Instances | File.InstanceTypes | File.Refs | File.DependObjs | File.DependKeys | File.DependUuidIndices>
-type Shared = Pick<IFileData, File.Version | File.SharedUuids | File.SharedStrings | File.SharedClasses | File.SharedMasks>;
-const PACKED_SECTIONS = File.Instances;
-interface IPackedFileData extends Shared {
-    [PACKED_SECTIONS]: IFileData[];
+type IFileData = MapEnum<{
+    [x in keyof IFileDataMap as `${x}`]: IFileDataMap[x];
+}, 11 /* Currently we should manually specify the enumerators count. */>;
+
+type IIntrudedFileDataMap = Omit<IFileDataMap, File.Version> & {
+    [File.Context]: FileInfo & DeserializeContext;
 }
 
+type IIntrudedFileData = MapEnum<{
+    [x in keyof IIntrudedFileDataMap as `${x}`]: IIntrudedFileDataMap[x];
+}, 11 /* Currently we should manually specify the enumerators count. */>;
+
+type IDeserializeInput = IFileData | IIntrudedFileData;
+
+type ISharedData = TupleSlice<IFileData, 1, 5>;
+
+type IPackedFileSection = [
+    ...document: TupleSlice<IFileData, 5>,
+];
+
+const PACKED_SECTIONS = File.Instances;
+
+type IPackedFileData = [
+    /** Version. */
+    version: number,
+
+    ...shared: ISharedData,
+
+    sections: IPackedFileSection[],
```

Comment:
Since `IPackedFileData` is now a tuple, we can not do:
```ts
[
  version: number,
  shared: /* ...*/,
  [PACKED_SECTIONS]: IPackedFileSection[] // Syntax error
]
````;